python/context/ball_trajectories.py

```diff
@@ -55,38 +58,30 @@ def _list_files(
     return paths
 
 
-def to_stamped_trajectory(
-    input: StateTrajectory, sampling_rate_seconds: float
-) -> StampedTrajectory:
+def to_stamped_trajectory(input: DurationTrajectory) -> StampedTrajectory:
+
+    """
+    Converts a Duration trajectory to a stamped trajectory.
+    """
 
-    size = len(input)
-    stamps = np.array([i * sampling_rate_seconds * 1e6 for i in range(size)], np.uint)
-    positions = np.array([np.array(s.position) for s in input])
+    durations = input[0]
+    positions = input[1]
+    size = len(durations)
+    stamps = np.zeros(size, np.uint)
+    stamps[1:] = stamps[1:] + np.cumsum(durations[:-1])
```

Comment:
Since `stamps` is initialised to zero, it should enough to just assign the result of `cumsum` here without the addition.